_blog/2018/soft-multitenancy.md

```diff
@@ -0,0 +1,319 @@
+---
+title: "Istio Multitenancy Support"
+overview: Using Kubernetes namespace and RBAC to create an Istio soft multitenancy environment
+publish_date: March 21, 2018
+subtitle: Using multiple Istio control planes and RBAC to create multitenancy
+attribution: John Joyce and Rich Curran
+
+order: 92
+
+layout: blog
+type: markdown
+redirect_from: "/blog/soft-multitenancy.html"
+---
+{% include home.html %}
+Multitenancy is commonly used in many environments across many different applications,
+but the implementation details and functionality provided on a per tenant basis does not
+follow one model in all environments.  The [Kubernetes multitenancy working group](
+https://github.com/kubernetes/community/blob/master/wg-multitenancy/README.md)
+is working to define the multitenant use cases and functionality that should be available
+within Kubernetes. However, from their work so far it is clear that only "soft multitenancy"
+is possible due to the inability to fully protect against malicious containers or workloads
+gaining access to other tenant's pods or kernel resources.
+
+## Defining soft multitenancy for Istio
+While discussing the strengths and benefits of deploying applications on top of Istio, it
+became apparent that there are multitenant use cases for Istio even within this Kubernetes
+"soft multitenancy" definition that does not provide absolute protection amongst tenants.
+A viable use case for this scenario is shared corporate infrastructure where malicious
+actions are not expected but a clean separation of the tenants is still required. A few
+different multitenant models can be considered.
+1.	A single mesh with multiple applications, one for each tenant on the mesh. The cluster
+administrator gets control and visibility mesh wide and across all applications, while the
+tenant administrator only gets control of a specific application.
+1.	A single Istio control plane with multiple meshes, one mesh per tenant. The cluster
+administrator gets control and visibility across the entire Istio control plane and all
+meshes, while the tenant administrator only gets control of a specific mesh.
+1.	A single Kubernetes control plane with multiple Istio control planes, one per tenant.
+The cluster administrator gets control and visibility across all the Istio control planes,
+while the tenant administrator only gets control of a specific Istio instance.
+1.	A single cloud environment (cluster controlled), but multiple Kubernetes control planes
+(tenant controlled).
+
+Options 1, 2 and 4 either can't be properly supported without code changes or don't fully
+address the use cases. Current Istio capabilities are well suited to providing option 3 so 
```

Comment:
What are the use cases? This is a very confusing section that uses a lot of lose terminology, e.g., scenarios, use cases, models, options, without clearly explaining what they're referring to. I would suggest either elaborate, or leave out all this option stuff and just focus on what the rest of the blog is about: one way to implement multi tenancy with Istio on top of kubernetes (i.e., option 3).

Reply:
agree with @frankbu - this section is a little confusing and could be dropped.